app/src/main/java/com/manichord/mgit/ssh/PrivateKeyGenerate.kt

```
@@ -0,0 +1,83 @@
+                .setPositiveButton(R.string.label_generate_key) { _, _ -> generateKey() }
+                .setNegativeButton(R.string.label_cancel) { _, _ -> }
+        return builder.create()
+    }
+
+    private fun generateKey() {
+        val newFilename = mNewFilename.text.toString().trim { it <= ' ' }
+        if (newFilename == "") {
+            showToastMessage(R.string.alert_new_filename_required)
+            mNewFilename.error = getString(R.string.alert_new_filename_required)
+            return
+        }
+        if (newFilename.contains("/")) {
+            showToastMessage(R.string.alert_filename_format)
+            mNewFilename.error = getString(R.string.alert_filename_format)
+            return
+        }
+        val keySize = mKeyLength.text.toString().toInt()
+        if (keySize < 1024) {
+            showToastMessage(R.string.alert_too_short_key_size)
+            mNewFilename.error = getString(R.string.alert_too_short_key_size)
+            return
+        }
+        if (keySize > 16384) {
+            showToastMessage(R.string.alert_too_long_key_size)
+            mNewFilename.error = getString(R.string.alert_too_long_key_size)
+            return
+        }
+        val type = when (mRadioGroup.checkedRadioButtonId) {
+            R.id.radio_dsa -> KeyPair.DSA
+            // JSCH doesn't support writing ED25519 keys yet, only reading
+            //R.id.radio_ed25519 -> KeyPair.ED25519
+            else -> KeyPair.RSA
+        }
+        val newKey = File(PrivateKeyUtils.getPrivateKeyFolder(), newFilename)
+        val newPubKey = File(PrivateKeyUtils.getPublicKeyFolder(), newFilename)
+        try {
+            val jsch = JSch()
+            val kpair = KeyPair.genKeyPair(jsch, type, keySize)
+            kpair.writePrivateKey(FileOutputStream(newKey))
+            kpair.writePublicKey(FileOutputStream(newPubKey), "mgit")
+            kpair.dispose()
+        } catch (e: Exception) {
+            //TODO
```

I think we can't just swallow the exception here. 
Perhaps just show a generic error message or throw an Exception that will then allow the user to submit a crash report via email? (now that Sentry crash reporting has been removed).